.github/workflows/build-test-distribute.yaml

```diff
@@ -30,22 +28,66 @@ jobs:
           make clean
       - run: |
           make check
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: "Check if should run on all arch/os combinations"
+        if: github.event_name == 'push' || contains(github.event.pull_request.labels.*.name, 'ci/run-full-matrix')
+        run: |
+          echo 'ENABLED_GOARCHES=arm64 amd64' >> $GITHUB_ENV
+          echo 'ENABLED_GOOSES=linux darwin' >> $GITHUB_ENV
+          # some steps don't support cross running, so write the actual arch down here.
+          ARCH=$(uname -m)
+          case $ARCH in
+            aarch64)
+            RUNNER_ARCH=arm64
+            ;;
+            x86_64)
+            RUNNER_ARCH=amd64
+            ;;
+            *)
+            echo "Unsupported arch '$ARCH'"
+            exit 1
+            ;;
+          esac
+          echo "RUNNER_ARCH=$RUNNER_ARCH" >> $GITHUB_ENV
+      - uses: actions/setup-go@v4
+        with:
+          go-version-file: go.mod
+      # GitHub actions does not share cache across multiple jobs,
+      # so we have to operate cache in each job and action file
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.kuma-dev
+          key: ${{ runner.os }}-devtools-${{ hashFiles('mk/dependencies/deps.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-devtools
+      - run: |
+          make dev/tools
       - run: |
           make build
       - run: |
           make -j build/distributions
       - run: |
-          make -j images
+          ENABLED_GOARCHES=$RUNNER_ARCH make -j images
       - run: |
-          make -j docker/save
-      - name: Pack artifact
+          ENABLED_GOARCHES=$RUNNER_ARCH make -j docker/save
+      - name: Cleanup redundant build outputs
         run: |
-          tar -czf build.tar.gz ./build
-      - name: Temporarily saving build output
+          find  ./build/artifacts-* -maxdepth 0 -type d | xargs -I % rm -rf %
+          find ./build/distributions/* -maxdepth 0 -type d | grep -v '/out' | xargs -I % rm -rf %
+          find  ./build/tools-* -maxdepth 0 -type d | xargs -I % rm -rf %
+          rm -rf ./build/oapitmp
+          rm -rf ./build/ebpf/
```

Comment:
Do we have a mirror of this in circleci?

Reply:
CircleCI has a builtin function and it lists all items needed to persist. This is a replacement to that.